### PR TITLE
Problem: bindings break all the time

### PR DESCRIPTION
**Pull request by dkliban**
Solution: build and test bindings continuosly

This patch adds another job to the Travis build matrix. This new build installs pulpcore and pulp_file
and then builds a client library for each. The client libraries are built using pulp-swagger-codegen,
a thin wrapper around swagger-codegen-cli. After building the bindings, this job installs them and runs
a script that performs the following:

- Create a File Remote
- Create a Repository
- Sync a Repository
- Monitor the sync task
- Create an artifact from a local file
- Create a FileContent from the artifact
- Add the new FileContent to a repository version
- Monitor the repo version creation task
- Create a FilePublisher
- Create a publication from the latest version of the repository
- Monitor the publish task
- Create a distribution pointint to the publication

fixes: #4123
https://pulp.plan.io/issues/4123

### CODE REVIEW

**Comment by dkliban** [comment]
These bindings only work on Python < 3.7 for now. We are experiencing a known bug[0]. There is already a patch on the master branch of swagger-codegen.

[0] https://github.com/swagger-api/swagger-codegen/pull/8401

File: .travis/script.sh [hunk 1]
```diff
@@ -33,6 +33,18 @@ if [ "$TEST" = 'docs' ]; then
   exit
 fi
 
+if [ "$TEST" = 'bindings' ]; then
+  cd ..
+  git clone https://github.com/pulp/pulp-swagger-codegen.git
+  cd pulp-swagger-codegen
+  sudo ./generate.sh pulpcore 3.0.0rc1 python
```
Comment (bmbouter) [inline .travis/script.sh]: it would be really cool if these versions weren't commited here in a way that required them to be updated.

Comment (rochacbruno) [reply .travis/script.sh]: If the [hub](https://hub.github.com/) package is installed then:

```bash
$ hub release           
3.0.0rc1
```
So

```bash
sudo ./generate.sh pulpcore $(hub release) python           
```

Comment (dkliban) [reply .travis/script.sh]: I updated pulp-swagger-codegen with the ability to discover the version of the installed plugin. https://github.com/pulp/pulp-swagger-codegen/pull/2

I updated this PR to use the new interface.

**Pull request merged**